components/tikv_util/src/memory.rs

```diff
@@ -200,60 +211,44 @@ impl MemoryQuota {
         self.capacity.load(Ordering::Relaxed)
     }
 
-    pub fn set_capacity(&self, capacity: usize) {
+    pub fn set_capacity(&self, mut capacity: usize) {
+        // Value bigger than isize::MAX just means unlimited,
+        // so replace it with isize::MAX to avoid overflow.
+        if capacity > isize::MAX as usize {
+            capacity = isize::MAX as usize;
+        }
         self.capacity.store(capacity, Ordering::Relaxed);
     }
 
     pub fn alloc_force(&self, bytes: usize) {
-        let mut in_use_bytes = self.in_use.load(Ordering::Relaxed);
-        loop {
-            let new_in_use_bytes = in_use_bytes + bytes;
-            match self.in_use.compare_exchange_weak(
-                in_use_bytes,
-                new_in_use_bytes,
-                Ordering::Relaxed,
-                Ordering::Relaxed,
-            ) {
-                Ok(_) => return,
-                Err(current) => in_use_bytes = current,
-            }
-        }
+        self.in_use.fetch_add(bytes as isize, Ordering::Relaxed);
```

Comment:
Another issue

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=69bef8ec9096033e89d90b3c6b86d16a

Reply:
But the previous implementation also has this issue that the value of `in_use` can overflow.

Reply:
The second alloc will return an error in the previous implementation.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5da24bccc3031d787b7541a841247663

---

It's okay that sometimes `in_use` is larger than `capacity`, but it's not okay that `alloc` continues returning success.